### ISSUE TITLE
Pass Commit and Rollback actions to other middlewares

### ISSUE THREAD

**Issue by MatthieuLemoine** [redux-offline/redux-offline#77]
Hi!

I'm using both ```redux-offline``` & ```redux-logger``` and I first tought that ```redux-offline``` wasn't working because the Commit and Rollback actions were not logged by the ```redux-logger``` middleware (I really scratched my head on this one). In fact, those actions are not passed through the middleware chain as next is only called on the Request action.

What do you think about passing the actions down through the middleware chain ?

PROS : 

- Better integration with other middlewares
- Better debugging experience

CONS : 

- Depending on the middlewares order not all will be called with those actions e.g if ```redux-logger``` is defined higher in the middleware chain than ```redux-offline``` it will not be called with the Commit and Rollback actions

Whether or not, I thiink it will be a good idea to add a note on that in the README to avoid other users to make the same mistake and to lose time as ```redux-logger``` is heavy used middleware.

I can make a PR if needed.

**Comment by MatthieuLemoine** [comment]
The problem is related to [redux#1051](https://github.com/reactjs/redux/issues/1051).

I found a workaround to make ```redux-offline``` working with others middlewares as ```redux-logger``` : 

```javascript
 // Bad
const store = createStore(rootReducer, initialState, compose(offline({ effect : api }), _getMiddlewares()));

// Good
const store = offline({ effect : api })(createStore)(rootReducer, initialState, _getMiddlewares());

function _getMiddlewares() {
  let middlewares = [
    thunk,
  ];

  if (process.env.NODE_ENV !== 'production') {
    middlewares = [...middlewares, logger];
  }

  return applyMiddleware(...middlewares);
}
```

**Comment by axxag** [comment]
@MatthieuLemoine worked for me as well, thanks!

```typescript
import { applyMiddleware, compose, createStore, Store } from 'redux'
import { createEpicMiddleware, EpicMiddleware } from 'redux-observable'
import { offline } from 'redux-offline'
import { offlineConfig } from '../configs/reduxOfflineConfig'
import { rootEpic, rootReducer } from './modules/root'

// tslint:disable-next-line:no-string-literal
const composeEnhancers = window['__REDUX_DEVTOOLS_EXTENSION_COMPOSE__'] && __DEV__ ? window['__REDUX_DEVTOOLS_EXTENSION_COMPOSE__']({
  shouldHotReload: false,
}) : compose

const epicMiddleware: EpicMiddleware<AppActions, AppState> = createEpicMiddleware(rootEpic)

const store: Store<AppState> =
  offline(offlineConfig)(createStore)(rootReducer, composeEnhancers(applyMiddleware(epicMiddleware)))

export default store

```

**Comment by bitsal** [comment]
@MatthieuLemoine @agentilela I tried your variants and it doesn't work with `redux-saga` middleware properly (when saga middleware fires an action, offline middleware doesn't know about it).

It started working only after we've configured it in the following way:
```
import { createStore, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { offline } from 'redux-offline';
import { createOfflineMiddleware } from 'redux-offline/lib/middleware';
...

const sagaMiddleware = createSagaMiddleware();
const offlineConfig = createOfflineConfig();
const store = offline(offlineConfig)(createStore)(
     rootReducer,
     compose(
         applyMiddleware(
             sagaMiddleware,
             createOfflineMiddleware(offlineConfig)
         )
     )
 );
```
If you look at the sources you can see that after such configuration middleware becomes like this:
```
offline, saga, offline
```
or `offline(saga(offline))`
I am not sure I understand the nature of it.

What do you think, guys, about it? Did we configure saga and offline middlewares incorrectly or it is really an issue in `redux-offline` in a case of many middlewares that are want to listen to each other?

**Comment by MatthieuLemoine** [comment]
@bitsal What is ```createOfflineMiddleware``` ?

**Comment by bitsal** [comment]
@MatthieuLemoine updated above
I've taken it from sources. Ideally, if it is really an issue it should be fixed inside `offline(...)`.

**Comment by wacii** [comment]
The `createOffline()` api gives you control over where the offline middleware is inserted in the middleware chain. You can read about it [here](https://github.com/redux-offline-team/redux-offline#choose-where-the-offline-middleware-is-added). 

If anyone still has trouble integrating with one of the libraries mentioned in the thread, feel free to open a new issue.

**Comment by bradserbuddy** [comment]
FWIW, @bitsal's solution worked for us, but the documented `createOffline()` API did not. If anyone would like to take a look, to troubleshoot, let me know. Thx.

**Comment by gusgard** [comment]
@bitsal Thanks for the solution!

**Comment by sorodrigo** [comment]
@bradserbuddy thanks for the input, I’ll try to inspect what’s going on and if there’s something else we can do, or add it to the roadmap maybe.

**Comment by wacii** [comment]
@sorodrigo The middleware returned by `createOffline()` is just `createOfflineMiddleware(offlineConfig)`. The same thing used in the solution here.

Assuming `createOffline()` is not fundamentally broken, I am not sure what we can do here without some code to work off of.

**Comment by tadeo** [comment]
[Updated createOffline doc link](https://github.com/redux-offline/redux-offline/blob/develop/docs/api/create-offline.md)
